atom/browser/atom_browser_context.cc

```diff
@@ -78,10 +78,10 @@ AtomBrowserContext::AtomBrowserContext(const std::string& partition,
   base::StringToInt(command_line->GetSwitchValueASCII(switches::kDiskCacheSize),
                     &max_cache_size_);
 
-  if (!base::PathService::Get(DIR_USER_DATA, &path_)) {
-    base::PathService::Get(DIR_APP_DATA, &path_);
+  if (!base::PathService::Get(DIR_USER_CACHE, &path_)) {
+    base::PathService::Get(DIR_CACHE, &path_);
     path_ = path_.Append(base::FilePath::FromUTF8Unsafe(GetApplicationName()));
-    base::PathService::Override(DIR_USER_DATA, path_);
+    base::PathService::Override(DIR_USER_CACHE, path_);
```

Comment:
The docs indicate that the intent is for `DIR_USER_DATA` and `DIR_USER_CACHE` to _both_ exist side-by-side, but this PR removes the code that sets the fallback value of `DIR_USER_DATA`. Is that intentional?

Reply:
At startup DIR-CACHE is set by default to DIR_USER_DATA, so the fallback may remain. The problem is when the userData is changed, the DIR_CACHE is not sync and the cache does not follow anymore, It is what @MarshallOfSound want to fix. I agree and try to find the best solution